build.gradle

```diff
@@ -8,6 +8,6 @@ repositories {
 
 dependencies {
     compileOnlyApi "org.immutables:value:2.8.8:annotations"
```

Comment:
```suggestion
    compileOnlyApi "org.immutables:value:2.9.0:annotations"
```

Reply:
What's the significance of "annotations" here? And is this one missed because it's first, or because of annotations?

Reply:
`annotations` is the artifact classifier, and its meaning is specific to the `org.immutables` project -- it could be any string to match the file format. In the maven repository format, there is one metadata file for the `org.immutables:value` artifact, and for each published version, there are potential additional files published alongside the primary artifact. These additional files can be referenced by adding a classifier.

Assuming a project is configured to resolve from the Maven Central repository,

- `org.immutables:value:2.9.0:annotations` refers to https://repo1.maven.org/maven2/org/immutables/value/2.9.0/value-2.9.0-annotations.jar
- `org.immutables:value:2.9.0` refers to https://repo1.maven.org/maven2/org/immutables/value/2.9.0/value-2.9.0.jar
- The metadata file that describes the primary artifact is https://repo1.maven.org/maven2/org/immutables/value/2.9.0/value-2.9.0.pom -- this same metadata file is reused for secondary artifacts

The metadata file for this version unfortunately has no machine-readable information about which classifiers are available for a given version, and the [`maven-metadata.xml` with a list of available versions](https://repo1.maven.org/maven2/org/immutables/value/maven-metadata.xml) doesn't have information about classifiers either, so any handling of classifiers would likely have to attempt to fetch files to determine if they exist.

Note that Gradle supports a few formats for specifying the classifier -- see the documentation for [Groovy buildscripts](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.dsl.DependencyHandler.html#N16EA1) and [Kotlin buildscripts](https://gradle.github.io/kotlin-dsl-docs/api/org.gradle.kotlin.dsl/-dependency-handler-scope/invoke.html) for format-specific variants.

Some Java projects which use the Gradle build tool publish a "Gradle module metadata" file in the repository as well, which provides more information about secondary artifacts. See [an example](https://repo1.maven.org/maven2/ca/stellardrift/contract-validator/1.0.1/contract-validator-1.0.1.module), or [the spec for this format](https://github.com/gradle/gradle/blob/bad3b98e60f4860497284cd2750a1d0b8edf2086/subprojects/docs/src/docs/design/gradle-module-metadata-latest-specification.md). However, this module metadata format is not as universally supported as the plain Maven metadata format is.

If you look at the most recent commit to this repository, I've removed use of the classifier and results are [as expected in this PR](https://github.com/zml2008/renovate-issue-5619-reproducer/commit/e2cd50b616462c38d16f71c652aa00a77b1062f5), so this appears to be an issue with the use of a classifier.